src/App/Controllers/Accounts.php

```diff
@@ -3,77 +3,202 @@
+                                </div>
+                            </div>
+                        </div>
+                    <?php
+                }
+            }
+        }
+    }
+
+    public function register(): void
+    {
 
-        if(isset($_POST['submit'])){
-            $email = htmlspecialchars(trim($_POST['email']));
-            $password = htmlspecialchars(trim($_POST['password']));
+        $username = htmlspecialchars(trim(filter_input(INPUT_POST, 'username')));
+        $email = htmlspecialchars(trim(filter_input(INPUT_POST, 'email')));
+        $password = htmlspecialchars(trim(filter_input(INPUT_POST, 'password')));
+        $passwordConfirm = htmlspecialchars(trim(filter_input(INPUT_POST, 'password_confirm')));
+
+        if (filter_input(INPUT_POST, 'submit') !== null){
 
             $errors = [];
 
-            if(empty($email) || empty($password)){
-                $errors['empty'] = "Tous les champs n'ont pas été remplis!";
-            }elseif($this->accountRepo->isAdmin($email,$password) == 0){
-                $errors['exist']  = "Accès refusé";
+            if(empty($username) || !preg_match('/^[\w_]+$/', $username)) {
+                $errors['username'] = "Votre pseudo est invalide";
+            }else{
+                $user = $this->accountRepo->isRegister($username);
+                if($user){
+                    $errors['exist'] = "Ce pseudo existe déjà";
+                }
+            }
+            if(empty($email) || !filter_var($email, FILTER_VALIDATE_EMAIL)) {
+                $errors['email'] = "Votre email est invalide";
+            }
+            if(empty($password) || $password != $passwordConfirm) {
+                $errors['mdp'] = "Vous devez rentrer un mot de passe valide";
             }
 
-            if(!empty($errors)){
-                ?>
-                <div class="card red">
-                    <div class="card-content white-text">
-                        <?php
-                        foreach($errors as $error){
-                            echo $error."<br/>";
-                        }
-                        ?>
-                    </div>
-                </div>
-                <?php
+            if(!empty($errors)) {
+
+                echo '<div class="container">
+                            <div class="card red">
+                                <div class="card-content white-text">';
+
+                                    foreach ($errors as $error) {
+                                    echo $error . "<br/>";
+                                    }
+                echo '</div></div>
+            </div>';
+
             }else{
-                $_SESSION['admin'] = $email;
-                $this->page->renderBack('dashboard');
+                //retourne un message pour signaler l'envoi dun mail afin de valider le compte et créer un mdp
+                $users = $this->accountRepo->registerUser();
```

Comment:
Créer un class pour la gestion du mail 

```suggestion
      new MailSender($host, $port, $options)
```